ammico/summary.py

```diff
@@ -169,99 +236,364 @@ def load_vqa_model(self):
         )
         return summary_vqa_model, summary_vqa_vis_processors, summary_vqa_txt_processors
 
-    def analyse_image(self):
+    def analyse_image(
+        self,
+        analysis_type: Optional[str] = None,
+        subdict: dict = {},
```

Comment:
if you want to init an empty dict as the default value, you should set it to "None" and then init as {} if it is None explicitly (instantiation of mutable default keyword arguments in function calls). But still, it should not be provided here at all since it is given at class instantiation already.